debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/dialect/sqlserver/BytesType.java

```diff
@@ -30,20 +29,17 @@ class BytesType extends AbstractBytesType {
     public static final BytesType INSTANCE = new BytesType();
 
     @Override
-    public String getDefaultValueBinding(DatabaseDialect dialect, Schema schema, Object value) {
-        return String.format(dialect.getByteArrayFormat(), ByteArrayUtils.getByteArrayAsHex(value));
+    public String getDefaultValueBinding(Schema schema, Object value) {
+        return String.format(getDialect().getByteArrayFormat(), ByteArrayUtils.getByteArrayAsHex(value));
     }
 
     @Override
-    public String getTypeName(DatabaseDialect dialect, Schema schema, boolean key) {
+    public String getTypeName(Schema schema) {
         final int columnSize = Integer.parseInt(getSourceColumnSize(schema).orElse("0"));
         if (columnSize > 0) {
-            return dialect.getTypeName(Types.VARBINARY, Size.length(columnSize));
+            return getDialect().getJdbcTypeName(Types.VARBINARY, Size.length(columnSize));
         }
-        else if (key) {
-            return dialect.getTypeName(Types.VARBINARY, Size.length(dialect.getMaxVarbinaryLength()));
-        }
-        return dialect.getTypeName(Types.VARBINARY);
+        return getDialect().getJdbcTypeName(Types.VARBINARY, Size.length(getDialect().getMaxVarbinaryLength()));
```

Comment:
This is a behavior change. For non-key columns, the column will now be constructed as `VARBINARY(8000)` rather than `VARBINARY(MAX)`. This changes the potential of storing 2GB of bytes to only 8000 bytes.

Reply:
Yes, that's why I asked you on Zulip y'day about the key field handling. I re-introduced key handling this morning.